parser_library/src/workspaces/processor_file_impl.h

```diff
@@ -40,26 +43,27 @@ class processor_file_impl : public virtual file_impl, public virtual processor_f
     // Used by the macro tracer.
     parse_result parse_no_lsp_update(parse_lib_provider&, analyzing_context ctx, const library_data) override;
 
-    // Returns true if parsing occured since this method was called last.
-    bool parse_info_updated() override;
-
     const std::set<std::string>& dependencies() override;
 
     const semantics::lines_info& get_hl_info() override;
     const lsp::feature_provider& get_lsp_feature_provider() override;
     const std::set<std::string>& files_to_close() override;
     const performance_metrics& get_metrics() override;
 
+    void erase_cache_of_opencode(const std::string& opencode_file_name) override;
+
 private:
-    std::unique_ptr<analyzer> analyzer_;
+    std::unique_ptr<analyzer> opencode_analyzer_;
+    const analyzer* last_analyzer_;
 
     bool parse_inner(analyzer&);
 
-    bool parse_info_updated_ = false;
     std::atomic<bool>* cancel_;
 
     std::set<std::string> dependencies_;
     std::set<std::string> files_to_close_;
```

Comment:
can this be impacted by different environments during processing? In other word will this "match" the content in the cache?

Reply:
The `dependencies_` and `files_to_close_` members are only used by files that are parsed as open codes, so they should not be affected by macro cache.